File: frontend/src/types/Book.ts
import type {Endowment} from "./Endowment.ts";
import type {Genre} from "./Genre.ts";
import type {Author} from "./Author.ts";

export interface Book {
  id: number | string;
  number?: number | string;
  title?: string;
  authors?: Author[];
  genre?: Genre;
  endowment?: Endowment | null;
  manuscript?: boolean;
  print?: boolean;
  volume?: number | string;
  comment?: string;
  editorComment?: string;
}

File: backend-net/src/model/Book.cs
namespace backend_net.model;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; }
    public Author[] Authors { get; set; }
    public Genre Genre { get; set; }
    public Endowment? Endowment { get; set; }
    public int Number { get; set; }
    public bool Manuscript { get; set; }
    public bool Print { get; set; }
    public string? Comment { get; set; }
    public string? EditorComment { get; set; }
    public string? Volume { get; set; }
}